requirements.txt

```diff
@@ -1,26 +1,46 @@
-Flask==0.10.1
+# These requirements are read directly into setup.py, so specify them
+# in the order that they should be in in setup.py
+
+### This section is for dependencies of the libraries ###
+# these libraries are the set listed by pipdeptree -f -w
+# that are dependencies of libraries listed in the next section
+
+Werkzeug==0.11.5
+MarkupSafe==0.23
+itsdangerous==0.24
+six==1.10.0
+pycrypto==2.6.1
+future==0.15.2
+Mako==1.0.4
+Beaker==1.8.0
+funcsigs==0.4
+alabaster==0.7.7
+cryptography==1.3.1
+idna==2.1
+pyasn1==0.1.9
+enum34==1.1.2
+ipaddress==1.0.16
+cffi==1.5.2
+pycparser==2.14
+
+### This section is for the actual libraries ###
+
+# Flask must come after all other requirements that have "flask" as a
+# prefix due to a setuptools bug.
 Flask-Cors==2.0.1
+Flask==0.10.1
 avro==1.7.7
-coverage==3.7.1
-flake8==2.3.0  # Set due to conflict with pep8 in version 2.4.0
-freezegun==0.3.6
-guppy==0.1.10
 humanize==0.5.1
-mock==1.0.0
-nose==1.3.7
-pep8==1.6.2
 pysam==0.8.4
 PyVCF==0.6.7
 PyYAML==3.11
 requests==2.7.0
-snakefood==1.4
-Sphinx==1.3.1
 oic==0.7.6
 pyOpenSSL==0.15.1
 lxml==3.4.4
-# Requirements for test OIDC provider.
-cherrypy==3.2.4
+# Requirements for test OIDC provider
```

Comment:
These should be in development only. The OIDC provider is purely for testing, we should absolutely not encourage people to use it in deployments.